File: examples/python/patterns/code-organization/child-context.py
from aws_durable_execution_sdk_python import durable_with_child_context


@durable_with_child_context
def process_order(child: DurableContext, order: dict) -> str:
    child.step(validate_step(order))
    child.step(charge_step(order))
    child.step(schedule_step(order))
    return "ok"


context.run_in_child_context(process_order(order), name="process-order")

File: examples/java/patterns/code-organization/child-context.java
context.runInChildContext("process-order", String.class, child -> {
    child.step("validate", ValidationResult.class,
        ctx -> validator.validate(order));
    Receipt receipt = child.step("charge", Receipt.class,
        ctx -> payments.charge(order.total(), order.cardToken()));
    child.step("schedule", String.class,
        ctx -> shipments.schedule(order.id(), order.address()));
    return "ok";
});
